python/hail/expr/functions.py

```diff
@@ -3484,3 +3484,115 @@ def get_sequence(reference_genome, contig, position, before=0, after=0) -> Strin
         raise TypeError("Reference genome '{}' does not have a sequence loaded. Use 'add_sequence' to load the sequence from a FASTA file.".format(reference_genome.name))
     return _func("getReferenceSequence({})".format(reference_genome.name), tstr,
                  contig, position, before, after)
+
+
+@typecheck(locus=expr_locus(), is_female=expr_bool, father=expr_call, mother=expr_call, child=expr_call)
+def mendel_error_code(locus, is_female, father, mother, child):
+    """Compute a Mendelian violation code for genotypes.
+
+    >>> father = hl.call(0, 0)
+    >>> mother = hl.call(1, 1)
+    >>> child1 = hl.call(0, 1)  # consistent
+    >>> child2 = hl.call(0, 0)  # Mendel error
+    >>> locus = hl.locus('2', 2000000)
+
+    >>> hl.mendel_error_code(locus, True, father, mother, child1).value
+    None
+
+    >>> hl.mendel_error_code(locus, True, father, mother, child2).value
+    7
+
+    Note
+    ----
+    Ignores call phasing, and assumes diploid and biallelic.
```

Comment:
Add "except in the case where the proband is male and the locus is HemiX or HemiY". I don't want users to think they need to recode their haploid calls as diploid for the sex chromosomes to work.

Reply:
good call!